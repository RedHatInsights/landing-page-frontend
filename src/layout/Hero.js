import React from 'react';

import { Bullseye, Title, Button } from '@patternfly/react-core';

import './Hero.scss';

const Hero = () => (
    <section className='pf-c-page__main-section ins-c-hero pf-m-fill'>
        <Bullseye className='pf-l-flex pf-m-column pf-m-align-items-center'>
            <Title className='ins-c-hero__title pf-m-spacer-lg'> Explore our open, multicloud tools </Title>
            <Title headingLevel='h2' className='pf-m-spacer-xl'> Discover Red Hat <sup>Â®</sup> software-as-a-service </Title>
            <Button onClick={ () => window.insights.chrome.auth.login() }> Log in to your Red Hat account</Button>
            <a href='https://www.redhat.com/en/customers'> Not a customer? </a>
        </Bullseye>
    </section>
);

export default Hero;
