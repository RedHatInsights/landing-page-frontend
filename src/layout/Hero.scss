@import '~@redhat-cloud-services/frontend-components-utilities/styles/_mixins';
@import '~@patternfly/patternfly/sass-utilities/_all';

.ins-c-hero {

    background: url("/apps/chrome/assets/images/fedora-background-left.svg");
    background-repeat: no-repeat;
    background-size: cover;

    h1.ins-c-hero__title { color: white; }

    h2.ins-c-hero__sub-title { color: var(--pf-global--Color--200); }

    @media screen and (min-width: $pf-global--breakpoint--xs) {
        @include rem('min-height', 400px);
    }

    @media screen and (min-width: $pf-global--breakpoint--md) {
        @include rem('min-height', 550px);
    }
}