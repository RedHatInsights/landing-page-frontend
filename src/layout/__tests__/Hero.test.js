import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hero from '../Hero';

describe('Hero component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Hero />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('click: Not a customer link', () => {
    const wrapper = shallow(<Hero />);
    expect(wrapper.find('.ins-c-hero__new-customer').first().props().href).toBe(
      'https://www.redhat.com/en/customers'
    );
  });
});
