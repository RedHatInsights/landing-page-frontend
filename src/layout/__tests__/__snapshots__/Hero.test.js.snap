// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render correctly 1`] = `
<PageSection
  className="ins-c-hero pf-m-fill pf-l-flex pf-m-align-items-center pf-m-justify-content-center"
>
  <article
    className="pf-l-flex pf-m-column pf-m-align-items-center"
  >
    <Title
      className="ins-c-hero__title pf-m-spacer-lg"
      headingLevel="h1"
      size="4xl"
    >
      Explore our open, multicloud tools
    </Title>
    <Title
      className="ins-c-hero__sub-title pf-m-spacer-xl"
      headingLevel="h2"
    >
      Discover Red Hat
      <sup
        className="ins-c-rball"
      >
        Â®
      </sup>
      software-as-a-service
    </Title>
    <Button
      className="ins-c-hero__login"
      onClick={[Function]}
    >
      Log in to your Red Hat account
    </Button>
    <Button
      className="ins-c-hero__new-customer"
      component="a"
      href="https://www.redhat.com/en/customers"
      isInline={true}
      variant="link"
    >
      Not a customer?
    </Button>
  </article>
</PageSection>
`;
